manifests/broker/service.pp

```diff
@@ -12,38 +12,46 @@
   $service_ensure  = $kafka::broker::service_ensure,
   $jmx_opts        = $kafka::broker::jmx_opts,
   $log4j_opts      = $kafka::broker::log4j_opts,
-  $opts            = $kafka::broker::opts
+  $heap_opts       = $kafka::broker::heap_opts,
+  $opts            = $kafka::broker::opts,
 ) {
 
   if $caller_module_name != $module_name {
     fail("Use of private class ${name} by ${caller_module_name}")
   }
 
+  $service_name = 'kafka'
+
   if $service_install {
     if $::service_provider == 'systemd' {
       include ::systemd
 
-      file { '/usr/lib/systemd/system/kafka.service':
-        ensure  => present,
+      file { "${service_name}.service":
+        ensure  => file,
+        path    => "/usr/lib/systemd/system/${service_name}.service",
```

Comment:
yep. I'm currently not sure how exactly the overwrites work. Not everything is possible here. I've to read up the docs.
https://wiki.archlinux.org/index.php/systemd
https://www.freedesktop.org/software/systemd/man/systemd.unit.html
https://www.freedesktop.org/software/systemd/man/systemd.service.html#